File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Reviews/ReviewsForm.cs
using Serenity.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Forms;

[FormScript("ECommerceWebsite.Reviews")]
[BasedOnRow(typeof(ReviewsRow), CheckNames = true)]
public class ReviewsForm
{
    public string Name { get; set; }
    public string Email { get; set; }
    public string Subject { get; set; }
    public string Description { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/ReviewsForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ReviewsForm {
    Name: StringEditor;
    Email: StringEditor;
    Subject: StringEditor;
    Description: StringEditor;
}

export class ReviewsForm extends PrefixedContext {
    static readonly formKey = 'ECommerceWebsite.Reviews';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ReviewsForm.init)  {
            ReviewsForm.init = true;

            var w0 = StringEditor;

            initFormType(ReviewsForm, [
                'Name', w0,
                'Email', w0,
                'Subject', w0,
                'Description', w0
            ]);
        }
    }
}